src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/messaging/PrisonerReceivedIntoPrisonEventService.kt

```diff
@@ -2,30 +2,61 @@ package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging
 
 import mu.KotlinLogging
 import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionAlreadyExistsException
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleAlreadyExistsException
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.InductionScheduleCalculationRule
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.dto.CreateInductionScheduleDto
+import uk.gov.justice.digital.hmpps.domain.learningandworkprogress.induction.service.InductionService
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.ADMISSION
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.RETURN_FROM_COURT
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.TEMPORARY_ABSENCE_RETURN
 import uk.gov.justice.digital.hmpps.educationandworkplanapi.app.messaging.AdditionalInformation.PrisonerReceivedAdditionalInformation.Reason.TRANSFERRED
+import java.time.LocalDate
+import java.time.ZoneId
 
 private val log = KotlinLogging.logger {}
 
 @Service
-class PrisonerReceivedIntoPrisonEventService {
-  fun process(inboundEvent: InboundEvent, additionalInformation: PrisonerReceivedAdditionalInformation) =
+class PrisonerReceivedIntoPrisonEventService(private val inductionService: InductionService) {
+  fun process(inboundEvent: InboundEvent, additionalInformation: PrisonerReceivedAdditionalInformation) {
     when (additionalInformation.reason) {
       ADMISSION -> {
-        log.info { "Processing Prisoner Received Into Prison Event with reason ${additionalInformation.reason}" }
-        // TODO - process prisoner admission event in respect of what PLP needs to do on this event
+        log.info { "Processing Prisoner Received Into Prison Event with reason ${additionalInformation.reason} and prisoner number ${inboundEvent.prisonNumber()}" }
+        try {
+          inductionService.createInductionSchedule(
+            CreateInductionScheduleDto(
+              deadlineDate = calculateDeadlineDate(inboundEvent),
+              prisonNumber = inboundEvent.prisonNumber(),
+              scheduleCalculationRule = InductionScheduleCalculationRule.NEW_PRISON_ADMISSION,
+            ),
+          )
+        } catch (inductionAlreadyExistsException: InductionAlreadyExistsException) {
+          log.info { "An induction already exists for prisoner with id ${inboundEvent.prisonNumber()}" }
+          // update existing deadline date
```

Comment:
I think this comment and todo needs to live in the catch block for InductionScheduleAlreadyExistsException 🤔 

If a prisoner already has an Induction (ie. InductionAlreadyExistsException), it means they have already had an induction (perhaps via their previous sentence) and what we need to do in this case is to kick off the Review process by creating / updating a ReviewSchedule (which we dont have yet)

The other catch block is the one you want I think. If a prisoner is admitted to prison and they already have an InductionSchedule, then that is the case by which we need to update the deadline date 👍 

But thinking about it a bit more .... I think maybe the service method `inductionService.createInductionSchedule` might need some fettling with ... 🤔 

I think the various scenarios are for prisoner received event are:
* Prisoner does not have an Induction or an InductionSchedule
  * They are a brand new prisoner, never been seen before. We need to create an InductionSchedule record for them
* Prisoner already has an Induction (whether they have an InductionSchedule is largely irrelevant)
  * They are a re-admitting / re-offending prisoner, and from when they were in prison before they had a PLP Induction
  * We need to treat this as a Review process, and update (or create) the prisoner's ReviewSchedule record (doesn't exist yet, is loosely in scope of the work I am currently doing)
* Prisoner does not have an Induction, but does have an InductionSchedule
  * They are a re-admitting / re-offending prisoner, and had an InductionSchedule from when they were in prison previously, but their actual Induction was never done
  * We need to update the deadline date of their InductionSchedule

I _think_ they are the scenarios, and I think we might need to change the service method in some way to support this (might be as easy and changing the order in which the exceptions are thrown 🤔 )